indexer/jobs/source_job/pg_source_job.py

```diff
@@ -0,0 +1,382 @@
+                )
+            else:
+                result = (
+                    session.query(table)
+                    .join(unnest_query, table.block_number == unnest_query.c.block_number)
+                    .order_by(*table.__query_order__)
+                    .all()
+                )
+
+        finally:
+            session.close()
+
+        return result
+
+    def _query_logs_filter(self, start_block, end_block, log_filter):
+        query_filter = None
+
+        if len(log_filter["address"]) > 0:
+            query_filter = or_(
+                query_filter, Logs.address.in_([bytes.fromhex(address[2:]) for address in set(log_filter["address"])])
+            )
+
+        if len(log_filter["topics"]) > 0:
+            query_filter = or_(
+                query_filter, Logs.topic0.in_([bytes.fromhex(topic0[2:]) for topic0 in set(log_filter["topics"])])
+            )
+
+        if len(log_filter["transaction_hash"]) > 0:
+            query_filter = or_(
+                query_filter,
+                Logs.transaction_hash.in_(
+                    [bytes.fromhex(transaction_hash[2:]) for transaction_hash in set(log_filter["transaction_hash"])]
+                ),
+            )
+
+        query_filter = and_(query_filter, Logs.block_number >= start_block, Logs.block_number <= end_block)
+
+        session = self._service.get_service_session()
+        try:
+            logs = session.query(Logs).filter(query_filter).order_by(*Logs.__query_order__).all()
+        finally:
+            session.close()
+        return logs
+
+    def _query_transactions_filter(self, start_block, end_block, transaction_filter):
+        query_filter = None
+
+        if len(transaction_filter["hash"]) > 0:
+            query_filter = or_(
+                query_filter,
+                Transactions.hash.in_(
+                    [bytes.fromhex(transaction_hash[2:]) for transaction_hash in set(transaction_filter["hash"])]
+                ),
+            )
+
+        if len(transaction_filter["from_address"]) > 0:
+            query_filter = or_(
+                query_filter,
+                Transactions.from_address.in_(
+                    [bytes.fromhex(from_address[2:]) for from_address in set(transaction_filter["from_address"])]
+                ),
+            )
+
+        if len(transaction_filter["to_address"]) > 0:
+            query_filter = or_(
+                query_filter,
+                Transactions.to_address.in_(
+                    [bytes.fromhex(to_address[2:]) for to_address in set(transaction_filter["to_address"])]
+                ),
+            )
+
+        query_filter = and_(
+            query_filter, Transactions.block_number >= start_block, Transactions.block_number <= end_block
+        )
+
+        session = self._service.get_service_session()
+        try:
+            transactions = (
+                session.query(Transactions).filter(query_filter).order_by(*Transactions.__query_order__).all()
```

Comment:
Same question here. Need to test performance

Reply:
Check whether enough index.

Reply:
Index has been checked.
I will add partition key "timestamp" to the query to enhance its performance.